### PR TITLE
Added method on GeometricTransform to calculate the fiducial registration error

### PR DESCRIPTION
**Pull request by effting**

### CODE REVIEW

File: src/odemis/util/transform.py [hunk 1]
```diff
@@ -431,6 +431,30 @@ def __call__(self, x):
         return self.apply(x)
     __call__.__doc__ = apply.__doc__
 
+    def fre(self, x, y):
+        """
+        Returns the RMS value of the fiducial error registration (FRE).
```
Comment (tepals) [inline src/odemis/util/transform.py:436]: Why use FRE and not the target registration error? If I google FRE, I get multiple results saying to use TRE.

> Fiducial Registration Error (FRE) is often used as a surrogate for the Target Registration Error (TRE) that is the quantity of clinical interest [4]. Unfortunately, it has been shown [5] that TRE and FRE are uncorrelated. FRE is thus an unreliable predictor of registration accuracy. (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4423533/)

Could you also add another sentence briefly explaining when to use this function. Something like:
>This returns a measure for the registration error between source and target frame.

File: src/odemis/util/transform.py [hunk 2]
```diff
@@ -431,6 +431,30 @@ def __call__(self, x):
...
+        x : (n, 2) array
+            Coordinates in the source reference frame.
+        y : (n, 2) array
+            Coordinates in the destination reference frame. Must be of same
+            dimensions as `x`.
+
+        Returns
+        -------
+        fre : float
+            The root mean squared fiducial registration error.
```
Comment (pieleric) [inline src/odemis/util/transform.py:449]: Following on Thera's comment, please indicate that  the float >= 0, and that "the smaller, the better the fitting".

File: src/odemis/util/test/transform_test.py [hunk 3]
```diff
@@ -601,6 +601,7 @@ def test_rigid_transform_from_pointset_known_values(self):
             tform = RigidTransform.from_pointset(src, dst)
             self.assertAlmostEqual(0., _angle_diff(rotation, tform.rotation))
             numpy.testing.assert_array_almost_equal(translation, tform.translation)
+            self.assertAlmostEqual(tform.fre(src, dst), 0)
```
Comment (pieleric) [inline src/odemis/util/test/transform_test.py:604]: Could you add a test case where .fre() doesn't return 0? 
Maybe it's just a matter of extending src & dst with some non matching values, and checking that FRE increases?

**Review by pieleric — CHANGES_REQUESTED** [review]

File: src/odemis/util/transform.py (hunk 1 above)
Comment (effting) [reply src/odemis/util/transform.py:436]: I updated the docstring to clarify

File: src/odemis/util/transform.py (hunk 2 above)
Comment (effting) [reply src/odemis/util/transform.py]: I updated the docstring.

File: src/odemis/util/test/transform_test.py (hunk 3 above)
Comment (effting) [reply src/odemis/util/test/transform_test.py:604]: I added a test in `SimilarityTransformKnownValues.test_similarity_transform_from_pointset_umeyama`

**Review by pieleric — APPROVED** [review]

**Review by tepals — APPROVED** [review]

**Pull request merged**